src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -777,13 +778,61 @@ void loop_inst::concatenated_memory_mapping::slice_mem(const int64_t num_iterati
         auto mem = sliced_mems[i];
         pointers_to_data[stride > 0 ? i : (num_iters - i - 1)] = reinterpret_cast<char*>(mem->lock(stream));
     }
+
     char* concat_data = reinterpret_cast<char*>(concatenated_mem->lock(stream, cldnn::mem_lock_type::read));
-    ov::reference::split(concat_data, concat_mem_shape, elem_size, axis, num_iters, pointers_to_data.data());
+
+    auto concate_layout = concatenated_mem->get_layout();
+    auto trait = format::traits(concate_layout.format);
+    if (format::is_blocked(concate_layout.format) || concate_layout.data_padding) {
+        // Be careful: ov::reference::split is extremely slow.
+        // If we encounter any case where this code path is executed, we need to optimize it
+        ov::reference::split(concat_data, concat_mem_shape, elem_size, axis, num_iters, pointers_to_data.data());
+    } else {
+        const size_t part_length = concat_mem_shape.at(axis) / num_iters;
+        auto output_shape = concat_mem_shape;
+        auto out_data = pointers_to_data.data();
+        output_shape[axis] = part_length;
+
+        ov::Coordinate lower_bounds(concat_mem_shape.size(), 0);
+        ov::Coordinate upper_bounds = output_shape;
+        auto& lb_at_axis = lower_bounds[axis];
+        auto& ub_at_axis = upper_bounds[axis];
+
+        const auto out_last = std::next(out_data, num_iters);
+        for (auto out_first = out_data; out_first != out_last; ++out_first) {
+            auto continuous_size = 1;
+            auto dims_order = trait._order;
+            auto target_axis = std::find(dims_order.begin(), dims_order.end(), axis);
+            for (auto iter = target_axis + 1 ; iter != dims_order.end() ; ++iter) {
+                const size_t idx = *iter;
+                continuous_size *= ((output_shape.size() > idx) ? output_shape[idx] : 1);
```

Comment:
currently, this code makes windows build error. please check it again. thanks.
`loop.cpp(808): error C2220: the following warning is treated as an error`
`loop.cpp(808): warning C4267: '*=': conversion from 'size_t' to 'int', possible loss of data`

Reply:
Cleared.